### PR TITLE
feat(plugin): jwt-token plugin allow user defined token parameter name

### LINKED ISSUE #5710: request help: Can the jwt plugin support custom authentication parameter names?
### Issue description

目前jwt plguin 参数名称是写死的，能不能跟hmac一样，通过配置修改参数名称。

$ curl http://127.0.0.1:9080/index.html?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJ1c2VyLWtleSIsImV4cCI6MTU2NDA1MDgxMX0.Us8zh_4VjJXF-TmR5f8cif8mBU7SuefPlpxhH0jbPVI -i
HTTP/1.1 200 OK
Content-Type: text/html
Content-Length: 13175

将jwt修改为其他参数。

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

### PR DESCRIPTION
**Pull request by shaobo322**
### What this PR does / why we need it:

fix #5710

### Pre-submission checklist:

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Hi @shaobo322, thanks for your contribution! And please update your PR's checklist 😄

File: apisix/plugins/jwt-auth.lua [hunk 1]
```diff
@@ -159,8 +160,8 @@ local function fetch_jwt_token(ctx)
 
         return token
     end
-
-    token = ctx.var.arg_jwt
+    local token_arg = "arg_" .. (conf.param or "jwt")
```
Comment (spacewander) [inline apisix/plugins/jwt-auth.lua:163]: There is no need to use `or "jwt"`. We already set up the default value.

File: apisix/plugins/jwt-auth.lua [hunk 2]
```diff
@@ -45,6 +45,7 @@ local consumer_schema = {
     properties = {
         key = {type = "string"},
         secret = {type = "string"},
+        param = {type = "string", default = "jwt"},
```
Comment (spacewander) [inline apisix/plugins/jwt-auth.lua:48]: What about using the name "token_name" instead?

Comment (shaobo322) [reply apisix/plugins/jwt-auth.lua:48]: ok

File: apisix/plugins/jwt-auth.lua [hunk 3]
```diff
@@ -170,7 +171,7 @@ local function fetch_jwt_token(ctx)
         return nil, err
     end
 
-    local val, err = cookie:get("jwt")
+    local val, err = cookie:get(conf.param or "jwt")
```
Comment (leslie-tsang) [inline apisix/plugins/jwt-auth.lua:174]: Ditto

**Review by leslie-tsang — CHANGES_REQUESTED** [review]

**Comment by liangliang4ward** [comment]
Why is this PR not merged into the master？

**Pull request closed without merging**